### PR TITLE
 change method signature to EncodeZeroX(src []byte) (encoded string)

### PR DESCRIPTION
**Pull request by epishovska**
change method signature to EncodeZeroX(src []byte) (encoded string)

### CODE REVIEW

File: cmd/mailchain/internal/http/handlers/pubkey.go [hunk 1]
```diff
@@ -75,7 +75,6 @@ func GetPublicKey(finders map[string]mailbox.PubKeyFinder) func(w http.ResponseW
 		w.Header().Set("Content-Type", "application/json")
 		_ = json.NewEncoder(w).Encode(GetPublicKeyResponseBody{
 			PublicKey:         encodedKey,
-			PublicKeyEncoding: encodingType,
```
Comment (epishovska) [inline cmd/mailchain/internal/http/handlers/pubkey.go]: @robdefeo should i remove `PublicKeyEncoding`?

File: internal/pubkey/encode.go [hunk 2]
```diff
@@ -22,11 +22,12 @@ import (
 
 // EncodeByProtocol selects the correct encoding method, then encodes the public key with it.
 func EncodeByProtocol(in []byte, protocol string) (encoded, encodingType string, err error) {
+	encodingType = encoding.TypeHex0XPrefix
```
Comment (robdefeo) [inline internal/pubkey/encode.go]: I would suggest moving this inside the case statement for each case. It then is clearer for someone looking at this in the future that `encodingType` can change based on protocol. Although its more code, it is clearer. Always best to be clear.

File: internal/address/encode.go [hunk 3]
```diff
@@ -22,9 +22,10 @@ import (
 
 // EncodeByProtocol takes an address as `[]byte` then selects the relevant encoding method to encode it as string.
 func EncodeByProtocol(in []byte, protocol string) (encoded, encodingType string, err error) {
+	encodingType = encoding.TypeHex0XPrefix
```
Comment (robdefeo) [inline internal/address/encode.go]: I would suggest moving this inside the case statement for each case. It then is clearer for someone looking at this in the future that `encodingType` can change based on protocol. Although its more code, it is clearer. Always best to be clear.

**Comment by robdefeo** [comment]
Thank you @epishovska for your first contribution!!! 😄 this fixes #395

**Pull request merged**